{{ cookiecutter.folder_name }}/.travis.yml

```diff
@@ -79,32 +90,51 @@ before_script:
 {% endif %}
 
 script:
-  - flake8 {{ cookiecutter.repo_name }}
-  - coverage run run_tests.py
-  - set -e
+  - |
+    if [[ $UNIT_TEST ]]; then
+      echo "Running tests"
+      coverage run run_tests.py
+    fi
+  - |
+    if [[ $UNIT_TEST ]]; then
+      coverage report -m
+    fi
   {% if cookiecutter.use_doctr == "yes" %}
   - |
     if [[ -n "$DOCTR_DEPLOY_ENCRYPTION_KEY_{{ cookiecutter.github_repo_group.upper().replace('-','_') }}_{{ cookiecutter.repo_name.upper() }}" && $BUILD_DOCS ]]; then
       # Create HTML documentation  
+      set -e
       pushd docs
       make html
       popd
       #Publish docs.
       doctr deploy . --built-docs docs/build/html --deploy-branch-name gh-pages --command "touch .nojekyll; git add .nojekyll"
+      set +e
     fi
    {% endif %}
 
 after_success:
-  - coverage report -m
-  - codecov
-
   - |
-    if [[ $TRAVIS_PULL_REQUEST == false && $TRAVIS_REPO_SLUG == $OFFICIAL_REPO ]]; then
-      if [[ $TRAVIS_BRANCH == $TRAVIS_TAG  && $TRAVIS_TAG != '' && $PCDS_CHANNEL == 'pcds-tag' ]]; then
-        export ANACONDA_API_TOKEN=$CONDA_UPLOAD_TOKEN_TAG
-        anaconda upload bld-dir/linux-64/*.tar.bz2
-      elif [[ $TRAVIS_BRANCH == 'master' && $TRAVIS_TAG == '' && $PCDS_CHANNEL == 'pcds-dev' ]]; then
-        export ANACONDA_API_TOKEN=$CONDA_UPLOAD_TOKEN_DEV
-        anaconda upload bld-dir/linux-64/*.tar.bz2
-      fi
+    if [[ $UPLOAD ]]; then
+      pip install codecov
+      echo "Uploading to codecov"
+      codecov
+    fi
+  - |
+    if [[ $UPLOAD && $TRAVIS_PULL_REQUEST == false && $TRAVIS_REPO_SLUG == $OFFICIAL_REPO && $TRAVIS_BRANCH == $TRAVIS_TAG  && $TRAVIS_TAG != '' && $CONDA_UPLOAD_TOKEN_TAG != '' ]]; then
+      echo "Installing anaconda-client"
+      conda activate base
+      conda install anaconda-client
+      echo "Uploading to pcds-tag channel"
+      export ANACONDA_API_TOKEN=$CONDA_UPLOAD_TOKEN_TAG
+      anaconda upload bld-dir/noarch/*.tar.bz2
+    fi
+  - |
+    if [[ $UPLOAD && $TRAVIS_PULL_REQUEST == false && $TRAVIS_REPO_SLUG == $OFFICIAL_REPO && $TRAVIS_BRANCH == 'master' && $TRAVIS_TAG == '' && $CONDA_UPLOAD_TOKEN_DEV != '' ]]; then
+      echo "Installing anaconda-client"
+      conda activate base
+      conda install anaconda-client
+      echo "Uploading to pcds-dev channel"
+      export ANACONDA_API_TOKEN=$CONDA_UPLOAD_TOKEN_DEV
+      anaconda upload bld-dir/noarch/*.tar.bz2
```

Comment:
Is this boilerplate enough that we would want to move it to the CI helpers? We could discuss this for a later commit so I'll just approve for now.

Reply:
I think so. I'd like to get @ZLLentz's help with that, as this is mostly his work I'm copying in here.

Reply:
Yes, this needs to transplant to CI helpers. I can help once I'm done making the motion ioc exist.